iroh/src/client.rs

```diff
@@ -323,6 +366,75 @@ where
     }
 }
 
+/// Outcome of a blob add operation.
+pub struct BlobAddOutcome {
+    /// The hash of the blob
+    pub hash: Hash,
+    /// The format the blob
+    pub format: BlobFormat,
+    /// The size of the blob
+    pub size: u64,
+    /// The tag of the blob
+    pub tag: Tag,
+}
+
+/// Progress stream for blob add operations.
+pub struct BlobAddProgress {
+    stream: Pin<Box<dyn Stream<Item = Result<AddProgress>> + Send + Unpin + 'static>>,
+}
+
+impl BlobAddProgress {
+    fn new(
+        stream: (impl Stream<Item = Result<impl Into<AddProgress>, impl Into<anyhow::Error>>>
+             + Send
+             + Unpin
+             + 'static),
+    ) -> Self {
+        let stream = stream.map(|item| match item {
+            Ok(item) => Ok(item.into()),
+            Err(err) => Err(err.into()),
+        });
+        Self {
+            stream: Box::pin(stream),
+        }
+    }
+    /// Finish writing the stream, ignoring all intermediate progress events.
+    ///
+    /// Returns a hash and a size. When importing a single blob, this is the hash and size of that
+    /// blob. When importing a collection, this is the hash of the collection and the total size of
+    /// all imported blobs.
```

Comment:
is the size including or excluding the size of the collection blob?

Reply:
It is excluding the size of the collection blob. Added this to the docs. Or should we change the behavior? Would need a little more work to report the collection blob size as well, but can do.